lib/bamboo/email.ex

```diff
@@ -203,7 +203,27 @@ defmodule Bamboo.Email do
   end
 
   @doc ~S"""
-  Adds an attachment to the email
+  Adds an data attachment to the email
+
+  ## Example
+    put_attachment(email, %Bamboo.Attachment{})
+
+  Requires the fields filename and data of the %Bamboo.Attachment{} struct to be set.
+
+  ## Example
+    def create(conn, params) do
+      #...
+      email
+      |> put_attachment(%Bamboo.Attachment{filname: "event.ics", data: "BEGIN:VCALENDAR..."})
+      #...
+    end
+  """
+  def put_attachment(%__MODULE__{attachments: attachments} = email, %Attachment{filename: _filename, data: _data} = attachment) do
```

Comment:
You actually can. Then the values are set to `nil` because we don't specify anything as enforced keys:

```
iex(1)> %Bamboo.Attachment{}
%Bamboo.Attachment{content_type: nil, data: nil, filename: nil, path: nil}
```

But you're right - since we don't check the actual value it doesn't make sense for that check. But it would only make sense to put pass an Attachment struct with both keys (`filename` and `data`) specified in there. Do you think we should throw an error if one of it is empty?